packages/react-dom/src/__tests__/ReactDOMSafariMicrotaskBug-test.js

```diff
@@ -27,9 +27,12 @@ describe('ReactDOMSafariMicrotaskBug-test', () => {
     window.queueMicrotask = function(cb) {
       queue.push(cb);
     };
-    simulateSafariBug = function() {
-      queue.forEach(cb => cb());
-      queue = [];
+    flushMicrotasksPrematurely = function() {
+      while (queue.length > 0) {
```

Comment:
This simulates the Safari behavior of trying to drain the queue until it's empty. So that we don't "fix" this by adding an infinite loop.

Proof:

```html
<body>
<script>
let busy = true
const callback = () => {
  if (busy) {
    console.error('noo im busy')
    queueMicrotask(callback)
  } else {
    console.log('im ok')
  }
}
queueMicrotask(callback);
console.log("will add iframe");
const iframe = document.createElement("iframe");
iframe.src = "localhost";
document.body.appendChild(iframe);
console.log("did add iframe");
busy = false
</script>
</body>
```

gets Safari infinitely stuck